Ficha9/grafo_Ndirigidos.c

```diff
@@ -0,0 +1,262 @@
+/*****************************************************************/
+/*   Grafo c/ matriz de adjacencias | PROG2 | MIEEC | 2018/19    */      
+/*****************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "grafo_Ndirigidos.h"
+
+/* cria grafo com n vertices */
+grafo* grafo_novo(int n)
+{
+    int v;
+    grafo* g;
+    
+    g = (grafo*)malloc(sizeof(grafo));
+    g->tamanho = n;
+
+    /* cria array de listas de adjacencias */
+    g->adjacencias = (int**)malloc(n * sizeof(int*));
+    for(v=0; v<n; v++)
+    {
+        g->adjacencias[v] = (int*)calloc(n, sizeof(int));
+    }
+
+    return g;
+}
+
+/* apaga grafo e liberta memoria */
+void grafo_apaga(grafo* g)
+{
+    if(g == NULL)
+        return;
+
+    if(g->adjacencias != NULL)
+    {
+        int v;
+        for (v = 0; v < g->tamanho; v++)
+            free(g->adjacencias[v]);
```

Comment:
nit: use blocks
```suggestion
        {    free(g->adjacencias[v]); }
```

Reply:
Got it.

Reply:
I have noticed now, that function is not mine. Look only to the functions with the comments like `/* alinea 1.1 */` ate the beginning.